src/routes/loginOAuth.js

```diff
@@ -40,4 +40,43 @@ server.post("/auth/login", async (req, res) => {
   }
 });
 
+server.post("/auth/changePassword", async (req, res) => {
+  const { username, oldPassword, newPassword } = req.body;
+  let encryptedPassword
+  await encrypt.encryptPassword(newPassword).then(encryptedPass => {
+    encryptedPassword = encryptedPass;
+  })
+
+  const user = await Users.findOne({ username });
```

Comment:
Let's try a little different approach for this endpoint in order to reduce it's complexity. Let's find the user searching by username and password so we can get rid of some if else statements down below and make it more readable.

```suggestion
  const user = await Users.findOne({ username,  password: encryptedPassword});
```